pkg/api/org_users.go

```diff
@@ -386,6 +389,17 @@ func (hs *HTTPServer) updateOrgUserHelper(c *contextmodel.ReqContext, cmd org.Up
 	if !c.OrgRole.Includes(cmd.Role) && !c.IsGrafanaAdmin {
 		return response.Error(http.StatusForbidden, "Cannot assign a role higher than user's role", nil)
 	}
+	if hs.Features.IsEnabled(featuremgmt.FlagOnlyExternalOrgRoleSync) {
+		// we do not allow to change role for external synced users
+		qAuth := login.GetAuthInfoQuery{UserId: cmd.UserID}
+		err := hs.authInfoService.GetAuthInfo(c.Req.Context(), &qAuth)
```

Comment:
<img width="1458" alt="image" src="https://user-images.githubusercontent.com/7727602/226910303-addf31d5-a92f-4928-a0e7-6b98f26d150a.png">

basic auth users should be able to change the role now. It is only applicable for external users. The code looks a bit ugly, will see if I could refactor before a new review

Reply:
potential refactor would be to allow the IsExternallySynced to take in a userID as a parameter instead. This would clean up this code and one could be able to use it with only the ID of the user without needing to know If there is a AuthModule attached.

leaving this as a potential refactor for later